### DISCUSSION TITLE
.NET CLR performance compared to Java JRE

### DISCUSSION THREAD

**Discussion by ygoe** [dotnet/runtime#78473] [Q&A]
Hello,

Around 10 years ago, everybody (maybe outside the .NET universe) told that the Java runtime would execute applications considerably faster than what .NET did then. They said something about JIT and hot paths and things I don't fully understand. I'm not a Java expert, I don't even like the language.

But where are we today? .NET is announced to become faster by a few percent on all levels with every release in recent years. Can it be compared somehow with the JRE? Have we surpassed it yet? Is the last reason to prefer Java somewhere now obsolete? I can't find any numbers on that topic. But nobody is publishing the old statements anymore either.

**Comment by gfoidl** [comment]
E.g.

* https://benchmarksgame-team.pages.debian.net/benchmarksgame/fastest/csharp.html
* https://programming-language-benchmarks.vercel.app/csharp-vs-java
* https://www.techempower.com/benchmarks

I left no comments, see the numbers yourself (but also check code if the implementation is representative).
